uvicorn/server.py

```diff
@@ -57,6 +57,8 @@ def __init__(self, config: Config) -> None:
 
     def run(self, sockets: Optional[List[socket.socket]] = None) -> None:
         self.config.setup_event_loop()
```

Comment:
I think you should call setup_event_loop as close to asyncio.run as possible

Reply:
In other words... 🤔